### PR TITLE
Reliability fix for secure context

### PR DESCRIPTION
**Pull request by caesar1995**
We will only create a new SafeDeleteContext_SECURITY and assign it to refContext when we are 100% sure that the current refContext is invalid and its backing native structure is invalid as well.

Fix internal bug 645542. 

/cc: @ChrisAhna @dotnet/ncl

### CODE REVIEW

File: src/Common/src/Interop/Windows/sspicli/SecuritySafeHandles.cs [hunk 1]
```diff
@@ -534,7 +534,11 @@ internal abstract partial class SafeDeleteContext : SafeHandle
 
                     if (refContext == null || refContext.IsInvalid)
                     {
-                        refContext = new SafeDeleteContext_SECURITY();
+                        // Previous versions unconditionally built a new "refContext" here, but would pass
+                        // incorrect arguments to InitializeSecurityContextW in cases where an "contextHandle" was
+                        // already present and non-zero.
+                        if (isContextAbsent)
```
Comment (rmkerr) [inline src/Common/src/Interop/Windows/sspicli/SecuritySafeHandles.cs]: I would merge this with the existing `if` statement.

**Comment by caesar1995** [comment]
@dotnet test Linux arm64 Release Build
@dotnet test Outerloop Linux x64 Debug Build
@dotnet test Outerloop Windows x64 Debug Build
@dotnet test Packaging All Configurations x64 Debug Build

**Comment by caesar1995** [comment]
@dotnet test Linux arm64 Release Build
@dotnet-bot test outerloop linux x64 debug build

**Comment by caesar1995** [comment]
Failure is in `System.Runtime.Serialization.Formatters.Tests`. Not related, merging.

**Pull request merged**